2-0-data-structures-and-algorithms/src/test/java/com/bobocode/stack/StackTest.java

```diff
@@ -1,88 +1,355 @@
+        intStack.push(74);
+
+        assertThat(contains(55)).isTrue();
+        assertThat(contains(59)).isTrue();
+        assertThat(contains(74)).isTrue();
+    }
+
+    @Test
+    @Order(6)
     void pushElementOntoEmptyStack() {
         intStack.push(243);
 
-        assertThat(intStack.pop()).isEqualTo(243);
+        assertThat(contains(243)).isTrue();
     }
 
     @Test
-    @Order(2)
+    @Order(7)
+    void pushElementToEmptyStackIsHead() {
+        intStack.push(10);
+
+        Object head = getHeadObject();
+        int innerElement = getNodeElementInt(head);
+
+        assertThat(innerElement).isEqualTo(10);
+    }
+
+    @Test
+    @Order(8)
+    void pushAnotherElementToStackIsHead() {
+        intStack.push(10);
+        intStack.push(15);
+        intStack.push(20);
+
+        Object head = getHeadObject();
+        int innerElement = getNodeElementInt(head);
+
+        assertThat(innerElement).isEqualTo(20);
+    }
+
+    @Test
+    @Order(9)
+    void pushPutHeadElementToNext() {
+        fillTestStack(10, 15, 20);
+
+        assertThat(getNodeElementInt(getHeadObject())).isEqualTo(20);
+
+        intStack.push(30);
+
+        Object head = getHeadObject();
+        Object next = getNodeNextObject(head);
+        int nextElement = getNodeElementInt(next);
+
+        assertThat(nextElement).isEqualTo(20);
+    }
+
+    @Test
+    @Order(10)
+    void pushThrowsExceptionIfElementIsNull() {
+        assertThatNullPointerException().isThrownBy(() -> intStack.push(null));
+    }
+
+    @Test
+    @Order(11)
     void popElementFromEmptyStack() {
         assertThrows(EmptyStackException.class, () -> intStack.pop());
     }
 
     @Test
-    @Order(3)
-    void pushElements() {
-        intStack = LinkedStack.of(23, 35, 72);
+    @Order(12)
+    void popElements() {
+        fillTestStack(55, 17, 66, 234);
 
-        intStack.push(55);
+        int lastElement = intStack.pop();
 
-        assertThat(intStack.pop()).isEqualTo(55);
+        assertThat(lastElement).isEqualTo(234);
     }
 
     @Test
-    @Order(4)
-    void popElements() {
-        intStack = LinkedStack.of(87, 53, 66);
+    @Order(13)
+    void popPreviousElementBecomesHead() {
+        fillTestStack(10, 15, 20);
+        Object head = getHeadObject();
+
+        assertThat(getNodeElementInt(head)).isEqualTo(20);
 
         intStack.pop();
-        intStack.push(234);
-        Integer lastElement = intStack.pop();
+        head = getHeadObject();
 
-        assertThat(lastElement).isEqualTo(234);
+        assertThat(getNodeElementInt(head)).isEqualTo(15);
     }
 
     @Test
-    @Order(5)
+    @Order(14)
+    void sizeOnEmptyStack() {
+        int actualSize = getInnerSize();
+
+        assertThat(actualSize).isEqualTo(0);
+    }
+
+    @Test
+    @Order(15)
     void size() {
-        intStack = LinkedStack.of(87, 53, 66);
+        fillTestStack(1, 5, 7);
 
-        int actualSize = intStack.size();
+        int actualSize = getInnerSize();
```

Comment:
If you name the test "size" you should test a corresponding method. Currently, your test passes when `size()` is not implemented in the exercise.